applications/models.py

```diff
@@ -143,6 +143,32 @@ def upload_resume(self, resume_file):
         self.resume_upload_date = now_in_utc()
         self.save()
 
+    @transition(
+        field=state,
+        source=AppStates.AWAITING_SUBMISSION_REVIEW.value,
+        target=RETURN_VALUE(AppStates.AWAITING_SUBMISSION_REVIEW.value, AppStates.AWAITING_PAYMENT.value))
+    def approve_submission(self):
+        """Approve application submission"""
+        return (AppStates.AWAITING_PAYMENT.value
+                if self.all_submissions_are_reviewed() else AppStates.AWAITING_SUBMISSION_REVIEW.value)
+
+    @transition(
+        field=state,
+        source=AppStates.AWAITING_SUBMISSION_REVIEW.value,
+        target=AppStates.REJECTED.value
+    )
+    def reject_submission(self):
+        """Reject application submission"""
+
+    def all_submissions_are_reviewed(self):
+        """
+        Check if the application is waiting for some submissions to be reviewed. This is used only if
+        application is awaiting submissions review
+        """
+        if self.state != AppStates.AWAITING_SUBMISSION_REVIEW.value:
```

Comment:
A couple things:
1. This function still won't give us the right value if the user has fewer submissions than there are application steps. Example: Bootcamp 1 has 2 application steps: a video interview and a quiz. So far a user has submitted neither of those. Right now your function would return `True` for that user since they have no submissions with a null review status (because they have no submissions at all). [This function](https://github.com/mitodl/bootcamp-ecommerce/blob/master/applications/api.py#L53-L60) checks for the number of steps and compares to the number of submissions. If it looks like you can use any of that same logic here, it might be useful to pull that out into a separate API function.
1. I don't think it's necessary to check for the application state here.

Reply:
In the issue it is specified that this api should be called only when the application is in "AWAITING_SUBMISSION_REVIEW` state. The application state can only be in `AWAITING_SUBMISSION_REVIEW` when all applications steps have submissions. That's why there is no reason to use the same logic as in that function, because the state of the application already implies that all the application steps have been submitted.

Reply:
We can only move on to `AWAITING_PAYMENT` when the user has (a) submitted something for all of the required steps, and (b) all of those submissions are approved. In your logic for `approve_submission`, it considers (b) but not (a) when deciding to transition the state to `AWAITING_PAYMENT`.

Reply:
A user can be in `AWAITING_SUBMISSION_REVIEW` multiple times. If the application has 2 steps, and the user submits something for the first step, approving that first submission should move them back to `AWAITING_USER_SUBMISSIONS` because they need to submit something for step 2.

Reply:
(Sorry, I really should have documented all of this more clearly 😬)

Reply:
thanks, for the explanation! I made a wrong assumption.

Reply:
Perfectly reasonable assumption given the documentation, which right now is just a bunch of docstrings. I just need to create a simple readme